docs/vocabulary.md

```diff
@@ -0,0 +1,289 @@
+third-party application or integration needed.
+
+Some of the core GitLab CI/CD terms are listed below. [1]
+
+- **Job**: Instructions that a runner has to execute.
+- **Stage**: Elements to group and run jobs in certain way - jobs in same
+stage are run in parallel and jobs in next stage run after the jobs from the
+previous stage completed successfully.
+- **Pipeline**: A collection of jobs split into different stages.
+- **Runner**: An agent or server that executes each job individually that can
+spin up or down as needed.
+- **Trigger**: Trigger allows you to define downstream pipeline trigger.
+
+### Jenkins
+
+[Jenkins](https://jenkins.io/) is an open source automation server which
+enables developers around the world to reliably build, test, and deploy their
+software.
+
+Some of the core Jenkins terms are listed below. [2]
+
+- **Project**: A user-configured description of work which Jenkins should
+perform, such as building a piece of software, etc.
+- **Job**: A deprecated term, synonymous with Project.
+- **Stage**: Stage is part of Pipeline, and used for defining a conceptually
+distinct subset of the entire Pipeline
+- **Pipeline**: A user-defined model of a continuous delivery pipeline.
+- **Node**: A machine which is part of the Jenkins environment and capable of
+executing Pipelines or Projects. Both the Master and Agents are considered to
+be Nodes.
+- **Trigger**: A criteria for triggering a new Pipeline run or Build.
+
+### Jenkins X
+
+[Jenkins X](https://jenkins-x.io/) provides pipeline automation, built-in
+GitOps, and preview environments to help teams collaborate and accelerate
+their software delivery at any scale.
+
+Some of the core Jenkins X terms are listed below. [3]
+
+- **Step**: Basic building block to execute one or more commands.
+- **Stage**: Collection of steps and the required configuration.
+- **Pipeline**: Collection of one or more stages to run.
+- **Agent**: Container image or Jenkins agent to schedule pipelines on.
+- **Trigger**: A pipeline trigger defines when to automatically run a pipeline.
+
+### Screwdriver
+
+[Screwdriver](https://screwdriver.cd) is an open source build platform designed
+for Continuous Delivery.
+
+Some of the core Screwdriver terms are listed below. [4]
+
+- **Step**: List of commands to execute.
+- **Job**: Collection of the steps to run.
+- **Workflow**: Collection and order of jobs to run.
+- **Pipeline**: Used interchangeably with workflow. (*citation needed*)
+- **Trigger**: A pipeline trigger defines when to automatically run a pipeline
+or job. (*citation needed*)
+
+### Spinnaker
+
+[Spinnaker](https://www.spinnaker.io/) is an open source, multi-cloud continuous
+delivery platform for releasing software changes with high velocity and
+confidence.
+
+Some of the core Spinnaker terms are listed below. [5]
+
+- **Stage**: An atomic building block for a pipeline, describing an action that
+the pipeline will perform.
```

Comment:
Stages are actually not atomic, but Tasks within a Stage are. A Stage can be comprised of Tasks or synthetic stages. For illustration: A `deploy` stage that targets two datacenters would actually be a composition of two synthetic `deploy` stages, with each having their own discrete Tasks to create the server groups, and so-on.

- **Task**: An atomic function to perform.
- **Stage**: A collection of sequential Tasks and composed Stages that describe a higher-level action the Pipeline will perform either linearly or in parallel.

Reply:
Thanks for the review Rob. I'll amend the PR, add task to the list and update the definition of stage as you suggested.

Btw, I took this from the Spinnaker concepts webpage so it could perhaps be updated with the details you provided.

https://www.spinnaker.io/concepts/#stage